WooCommerce/Classes/ViewRelated/Dashboard/DashboardViewController.swift

```diff
@@ -315,6 +318,24 @@ private extension DashboardViewController {
         navigationItem.title = Localization.title
     }
 
+    func configureShareButton() {
+        // Only show share button if the site is public.
+        guard let site = ServiceLocator.stores.sessionManager.defaultSite, site.isPublic else {
+            return
+        }
+        navigationItem.rightBarButtonItem = shareButton
+    }
+
+    @objc
+    func shareStore() {
```

Comment:
Nit: This logic could be moved to view model as well.

Reply:
Also updated in 1ec15ac06953d3812128b3470cc392622928020b.